std::set_unexpected

   From cppreference.com
   < cpp | error

   C++
   Compiler support
   Freestanding and hosted
   Language
   Standard library
   Standard library headers
   Named requirements
   Feature test macros (C++20)
   Language support library
   Concepts library (C++20)
   Metaprogramming library (C++11)
   Diagnostics library
   General utilities library
   Strings library
   Containers library
   Iterators library
   Ranges library (C++20)
   Algorithms library
   Numerics library
   Localizations library
   Input/output library
   Filesystem library (C++17)
   Regular expressions library (C++11)
   Concurrency support library (C++11)
   Technical specifications
   Symbols index
   External libraries

   Diagnostics library
   Exception handling
   exception
   uncaught_exceptionuncaught_exceptions
   (until C++20*)(C++17)
   exception_ptr
   (C++11)
   make_exception_ptr
   (C++11)
   current_exception
   (C++11)
   rethrow_exception
   (C++11)
   nested_exception
   (C++11)
   throw_with_nested
   (C++11)
   rethrow_if_nested
   (C++11)
   Exception handling failures
   terminate
   terminate_handler
   get_terminate
   (C++11)
   set_terminate
   bad_exception
   unexpected
   (until C++17*)
   unexpected_handler
   (until C++17*)
   get_unexpected
   (C++11)(until C++17*)
   set_unexpected
   (until C++17*)
   Error codes
   Error codes
   errno
   Exception categories
   logic_error
   invalid_argument
   domain_error
   length_error
   out_of_range
   runtime_error
   range_error
   overflow_error
   underflow_error
   tx_exception
   (TM TS)
   System error support
   error_category
   (C++11)
   generic_category
   (C++11)
   system_category
   (C++11)
   error_condition
   (C++11)
   errc
   (C++11)
   error_code
   (C++11)
   system_error
   (C++11)
   Assertions
   assert
   Stacktrace
   stacktrace_entry
   (C++23)
   basic_stacktrace
   (C++23)

   Defined in header <exception>
   std::unexpected_handler set_unexpected( std::unexpected_handler f )
   throw();
   (until C++11)
   std::unexpected_handler set_unexpected( std::unexpected_handler f )
   noexcept;
   (since C++11)
   (deprecated)
   (removed in C++17)

   Makes f the new global std::unexpected_handler and returns the
   previously installed std::unexpected_handler.

   This function is thread-safe. Every call to std::set_unexpected
   synchronizes-with (see std::memory_order) the subsequent calls to
   std::set_unexpected and std::get_unexpected
   (since C++11)

Parameters

   f - pointer to function of type std::unexpected_handler, or null
   pointer

Return value

   The previously-installed unexpected handler, or a null pointer value if
   none was installed.

See also

   unexpected
   (removed in C++17)
   function called when dynamic exception specification is violated
   (function)
   get_unexpected
   (C++11)(removed in C++17)
   obtains the current unexpected_handler
   (function)
   unexpected_handler
   (removed in C++17)
   the type of the function called by std::unexpected
   (typedef)
   Retrieved from
   "https://en.cppreference.com/mwiki/index.php?title=cpp/error/set_unexpe
   cted&oldid=108415"

Navigation

     * Online version
     * Offline version retrieved 2024-06-10 08:17.

     * This page was last modified on 8 December 2018, at 20:37.
     * This page has been accessed 43,918 times.
